File: bro-agent-api/src/main/python/bro_agent/agent/FinishedActionState.py
import bro_agent.agent.states


# Final state of an action
# Awaits cancel message or new action message
class FinishedActionState(bro_agent.agent.WaitingForActionState.WaitingForActionState):
    """
    Final state of an action
    Awaits cancel message or new action message
    """
    def __init__(self, agent, backup_name, action):
        # @param agent The agent participating in action
        # @param backupName backup name
        # @param action action type
        bro_agent.agent.WaitingForActionState.WaitingForActionState.__init__(self, agent)
        self.action = action
        self.backup_name = backup_name

    def process_message(self, message):
        """ Process an incoming message"""
        if self.is_cancel_message(message) is True:
            return bro_agent.agent.CancelActionState.CancelActionState(self.agent, self.backup_name, self.action)
        return bro_agent.agent.WaitingForActionState.WaitingForActionState.process_message(self, message)

File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/agent/FinishedActionState.java
/*------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *----------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.agent;

import com.ericsson.adp.mgmt.action.Action;
import com.ericsson.adp.mgmt.control.OrchestratorControl;

/**
 * Final state of an action
 * Awaits cancel message or new action message
 */
public class FinishedActionState extends WaitingForActionState {

    private final String backupName;
    private final Action action;

    /**
     * @param agent The agent participating in action
     * @param backupName backup name
     * @param action action type
     */
    public FinishedActionState(final Agent agent, final String backupName, final Action action) {
        super(agent);
        this.action = action;
        this.backupName = backupName;
    }

    @Override
    public AgentState processMessage(final OrchestratorControl message) {
        if (isCancelMessage(message)) {
            return new CancelActionState(this.agent, this.backupName, this.action);
        }
        return super.processMessage(message);
    }

}
